### ISSUE TITLE
Suppression of no-member false positives broken

### ISSUE THREAD

**Issue by bkline** [PyCQA/pylint#5625]
### Bug description

```python
#!/usr/bin/env python3

class Repro:
    def __init__(self, **opts):
        for name in ("name", "error", "created", "active", "dtd"):
            setattr(self, name, opts.get(name))

    def show_error(self):
        # pylint: disable=no-member
        print(f"{self.name} error: {self.error}")

    def __str__(self):
        # pylint: disable=no-member
        return f"""\
     Name: {self.name}
  Created: {self.created}
   Active: {self.active}
      DTD: {self.dtd}
    Error: {self.error}
"""


opts = dict(name="repro", created="2021-12-31", active=True, dtd="repro.dtd")
instance = Repro(**opts)
print(instance)
instance.show_error()
```

### Configuration

No custom configuration file used.

### Command used

```shell
pylint -E disable-no-member-bug-repro.py
```

### Pylint output

```shell
************* Module disable-no-member-bug-repro
disable-no-member-bug-repro.py:15:28: E1101: Instance of 'Repro' has no 'name' member (no-member)
disable-no-member-bug-repro.py:16:28: E1101: Instance of 'Repro' has no 'created' member (no-member)
disable-no-member-bug-repro.py:17:28: E1101: Instance of 'Repro' has no 'active' member (no-member)
disable-no-member-bug-repro.py:18:28: E1101: Instance of 'Repro' has no 'dtd' member (no-member)
disable-no-member-bug-repro.py:19:28: E1101: Instance of 'Repro' has no 'error' member (no-member)
```

### Expected behavior

No output is expected. What would _really_ be nice is for `pylint` to understand how `setattr()` works, but that's not the bug which this issue is reporting. It appears from the output that the directive to suppress the reporting of the `no-member` false positive is honored (no error is reported for line 10) some of the time, but is ignored inside so-called "dunder" methods (such as `__str__()`). Or at least, that's my guess for why it works in one place and not the other. It's possible that whoever actually tracks down the source for the bug may discover that the controlling difference for when the tool behaves correctly and when it does not is something else, possibly more abstruse. My original speculation was that it was the difference in the string delimiters, but I disproved that theory by putting the same string with the triple-quote delimiters inside the `show_error()` method, and observing that the `disable=no-member` directive was honored in that method.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.1 (main, Dec  6 2021, 23:20:29) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

Not relevant, I shouldn't think, as I can reproduce the bug on MacOS, Linux, and Windows boxes.

### Additional dependencies

None.

**Comment by bkline** [comment]
OK, I've got some more clues. It turns out that it's not the difference between `__whatever__()` methods and `whatever()` methods. As far as I have been able to determine so far, the bug is in play, regardless of whether or not the method is a "dunder" method, _iff_ both of the following two conditions are met:

1. the member reference is inside a string coded using multiline syntax
2. the string is part of a `return` statement

So, `pylint` behaves correctly for this method:
```python
    def __str__(self):
        # pylint: disable=no-member
        return f"name: {self.name}\ncreated: {self.created}\n"
```
but ignores the directive in this version:
```python
    def __str__(self):
        # pylint: disable=no-member
        return f"""\
name: {self.name}
created: {self.created}
"""
```
Same value, but different behavior in `pylint`. But then, introducing a temporary variable, to avoid returning the string directly, we avoid the bug:
```python
    def __str__(self):
        # pylint: disable=no-member
        temp = f"""\
name: {self.name}
created: {self.created}
"""
        return temp
```
So it turns out my original hunch wasn't as far off target as I thought. It wasn't the delimiters, though, as I first suspected, but the multi-line strings enabled by the triple-quote delimiters, which were a necessary part of the conditions required in order to trigger the bug.

**Comment by DanielNoord** [comment]
I can no longer reproduce this on `main` and think we have fixed this unknowingly 🎉
